components/compositing/compositor.rs

```diff
@@ -1143,11 +1296,35 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
             })
     }
 
-    fn create_pipeline_details_for_frame_tree(&mut self, frame_tree: &SendableFrameTree) {
-        self.pipeline_details(frame_tree.pipeline.id).pipeline = Some(frame_tree.pipeline.clone());
+    fn update_pipeline_details_tree(
+        &mut self,
+        frame_tree: &SendableFrameTree,
+        parent_pipeline_id: Option<PipelineId>,
+    ) {
+        let pipeline_id = frame_tree.pipeline.id;
+        let pipeline_details = self.pipeline_details(pipeline_id);
+        pipeline_details.pipeline = Some(frame_tree.pipeline.clone());
+        pipeline_details.parent_pipeline_id = parent_pipeline_id;
 
         for kid in &frame_tree.children {
-            self.create_pipeline_details_for_frame_tree(kid);
+            self.update_pipeline_details_tree(kid, Some(pipeline_id));
+        }
+    }
+
+    fn remove_pipeline_details_tree(&mut self, pipeline_id: PipelineId) {
```

Comment:
```suggestion
    fn remove_pipeline_details_recursively(&mut self, pipeline_id: PipelineId) {
```

Reply:
done in done in b2528dc1bbe43ea18822c51cc3c2d6c10222bc0e

Reply:
done in b2528dc1bbe43ea18822c51cc3c2d6c10222bc0e